source/Conversation.cpp

```diff
@@ -434,46 +452,122 @@ const Sprite *Conversation::Scene(int node) const
 
 
 // Find out where the conversation goes if the given option is chosen.
-int Conversation::NextNode(int node, int choice) const
+int Conversation::NextNode(int node, int element) const
 {
-	if(static_cast<unsigned>(node) >= nodes.size()
-			|| static_cast<unsigned>(choice) >= nodes[node].data.size())
+	if(!NodeIsValid(node) || !ElementIsValid(node, element))
 		return DECLINE;
 
-	return nodes[node].data[choice].second;
+	return nodes[node].elements[element].next;
+}
+
+
+
+// Returns whether the given node should be displayed.
+bool Conversation::ShouldShowText(const map<string, int64_t> &vars, int node, int element) const
+{
+	if(!NodeIsValid(node))
+		return false;
+	else if(IsChoice(node) ? !ElementIsValid(node, element) : element != 0)
+		return false;
+	const auto &data = nodes[node].elements[element];
+	if(data.conditions.IsEmpty())
+		return true;
+	else
+		return data.conditions.Test(vars);
```

Comment:
Seems like we can ditch the `else`s here, since it's going to fall through to the next check regardless.
```suggestion
	if(IsChoice(node) ? !ElementIsValid(node, element) : element != 0)
		return false;

	const auto &data = nodes[node].elements[element];
	if(data.conditions.IsEmpty())
		return true;
	return data.conditions.Test(vars);
```

Reply:
I like to leave `else`s in in cases like this, to make it more explicit to the reader. But that's purely a matter of style.

I'll snip all three `else`s next time I'm in a position to do so.